meta/store.go

```diff
@@ -735,12 +734,19 @@ func (s *Store) serveExecListener() {
 		// Accept next TCP connection.
 		var err error
 		conn, err := s.ExecListener.Accept()
-		if err != nil {
-			if strings.Contains(err.Error(), "connection closed") {
+		if opErr, ok := err.(*net.OpError); ok && opErr.Temporary() {
+			s.Logger.Printf("exec listener temporary accept error: %s", err)
+
+			select {
+			case <-s.closing:
```

Comment:
I would expect that if a signal is sent on `closing` then the `ExecListener` would be closed from somewhere else and a non-temporary error would result. I might be wrong though.